alphadia/libtransform.py

```diff
@@ -602,6 +604,58 @@ def forward(self, input: SpecLibBase) -> SpecLibBase:
         return input
 
 
+class MultiplexLibrary(ProcessingStep):
+    def __init__(self, multiplex_mapping: dict) -> None:
+        """Initialize the MultiplexLibrary step."""
+
+        self._multiplex_mapping = multiplex_mapping
+
+    def validate(self, input: str) -> bool:
+        """Validate the input object. It is expected that the input is a path to a file which exists."""
+        valid = True
+        valid &= isinstance(input, SpecLibBase)
+
+        # check if all modifications are valid
+        for _, channel_multiplex_mapping in self._multiplex_mapping.items():
+            for key, value in channel_multiplex_mapping.items():
+                for mod in [key, value]:
+                    print(mod)
+                    if mod not in MOD_DF.index:
+                        logger.error(f"Modification {mod} not found in input library")
+                        valid = False
+
+        return valid
+
+    def forward(self, input: SpecLibBase) -> SpecLibBase:
+        """Apply the MultiplexLibrary step to the input object."""
+
+        if "channel" in input.precursor_df.columns:
+            channel_unique = input.precursor_df["channel"].unique()
+            logger.info(f"Library already multiplexed for channel {channel_unique}")
+            input.precursor_df = input.precursor_df[input.precursor_df["channel"] == 0]
+
```

Comment:
The implicit assumption is that unmodified is always encoded as 0 (and not as "ref" of whatever)? Maybe then specify this also in the yaml, which relates to the comment above

Reply:
Is the case that a library e.g. has only channel 2 likely, and would the code behave as expected in this case?